pkg/cloudfabric-controller/controllerframework/controller_framework.go

```diff
@@ -338,57 +265,31 @@ func (c *ControllerBase) updateCachedControllerInstances(newControllerInstanceMa
 		klog.V(4).Infof("Controller %s instance %s mux released", c.controllerType, c.controllerName)
 	}()
 
-	sortedNewControllerInstancesLocal := SortControllerInstancesByKeyAndConvertToLocal(newControllerInstanceMap)
+	sortedNewControllerInstancesLocal := sortControllerInstancesByKeyAndConvertToLocal(newControllerInstanceMap)
 
 	// Compare
-	isUpdated, isSelfUpdated, newLowerBound, newUpperbound, newPos := c.tryConsolidateControllerInstancesLocal(sortedNewControllerInstancesLocal)
+	isUpdated, isSelfUpdated, newLowerBound, newUpperbound, newPos, newSortedControllerInstanceLocal :=
+		c.tryConsolidateControllerInstancesLocal(sortedNewControllerInstancesLocal)
 	if !isUpdated && !isSelfUpdated {
 		return
 	}
 
-	isSelfRangeUpdated := false
-
-	if isUpdated {
-		defer func() {
-			c.curPos = newPos
-			c.sortedControllerInstancesLocal = sortedNewControllerInstancesLocal
-			c.controllerInstanceMap = newControllerInstanceMap
-
-			if isSelfUpdated {
-				c.controllerKey = c.sortedControllerInstancesLocal[c.curPos].controllerKey
-			}
-
-			if isSelfRangeUpdated {
-				ResetFilterHandler(c, newLowerBound, newUpperbound)
-			}
-
-			c.IsDoneProcessingCurrentWorkloads()
-		}()
-	}
-
 	if isSelfUpdated {
 		var currentControllerInstance controllerInstanceLocal
-		if c.curPos < 0 {
-			currentControllerInstance = sortedNewControllerInstancesLocal[newPos]
-			isSelfRangeUpdated = true
+		if c.curPos < 0 { // new instance during registration process
+			currentControllerInstance = newSortedControllerInstanceLocal[newPos]
 		} else {
 			currentControllerInstance = c.sortedControllerInstancesLocal[c.curPos]
-
-			if currentControllerInstance.lowerboundKey != newLowerBound || currentControllerInstance.controllerKey != newUpperbound {
-				isSelfRangeUpdated = true
-			}
-		}
-
-		if isSelfRangeUpdated {
-			klog.Infof("Self range updated %d, %d", newLowerBound, newUpperbound)
 		}
 
 		message := fmt.Sprintf("Controller %s instance %s", c.controllerType, c.controllerName)
 		if currentControllerInstance.lowerboundKey != newLowerBound || currentControllerInstance.controllerKey != newUpperbound {
-			if c.sortedControllerInstancesLocal[c.curPos].lowerboundKey == 0 {
-				message += fmt.Sprintf(" old range [%d, %d]", c.sortedControllerInstancesLocal[c.curPos].lowerboundKey, c.sortedControllerInstancesLocal[c.curPos].controllerKey)
-			} else {
-				message += fmt.Sprintf(" old range (%d, %d]", c.sortedControllerInstancesLocal[c.curPos].lowerboundKey, c.sortedControllerInstancesLocal[c.curPos].controllerKey)
+			if c.curPos >= 0 {
```

Comment:
Shall the code just return if c.curPos < 0, as it means the registration is incomplete?

Reply:
> Shall the code just return if c.curPos < 0, as it means the registration is incomplete?

Yes, this code is also used in registration process. It needs to handle curPos < 0